cmd/main.go

```diff
@@ -103,6 +113,14 @@ func main() {
 	if err := agent.Listen(agent.Options{}); err != nil {
 		logger.Errorw("Failed to start gops agent", "error", err)
 	}
+	http.HandleFunc("/healthz", healthHandler)
+
+	errChan := make(chan error, 1)
+
+	go func() {
+		errChan <- http.ListenAndServe(":8080", nil)
+		logger.Error(errChan, "healthz check server failed")
```

Comment:
1. You can't just silently let the goroutine exit silently in this case.
2. We switched the whole codebase to Zap. Use the correct method signature.
```suggestion
		logger.Fatal("healthz check server failed", zap.Error(errChan))
```

Reply:
Oh wait, I misread this.
`ListenAndServe` returns an `error`, and you're writing this error into `errChan`.
In that case, just get the error returned by `ListenAndServe` and both write it into `errChan` and pass it to `Error` as an Error field.